File: apps/wearos/app/src/test/java/dev/bishnoi/forgelog/wear/logic/ElapsedTest.kt
package dev.bishnoi.forgelog.wear.logic

import org.junit.Assert.assertEquals
import org.junit.Test

class ElapsedTest {

    @Test
    fun `formats elapsed time as hours minutes and seconds`() {
        assertEquals("00:00:00", formatElapsed(0))
        assertEquals("00:00:09", formatElapsed(9))
        assertEquals("00:01:05", formatElapsed(65))
        assertEquals("00:59:59", formatElapsed(3599))
        assertEquals("01:00:00", formatElapsed(3600))
        assertEquals("01:01:01", formatElapsed(3661))
    }

    @Test
    fun `negative durations clamp to zero`() {
        assertEquals("00:00:00", formatElapsed(-5))
    }
}

File: apps/mobile/src/domain/__tests__/elapsed.test.ts
import { formatElapsed } from '../elapsed';

test('formats elapsed workout time as hh:mm:ss', () => {
  expect(formatElapsed(0)).toBe('00:00:00');
  expect(formatElapsed(9)).toBe('00:00:09');
  expect(formatElapsed(65)).toBe('00:01:05');
  expect(formatElapsed(3599)).toBe('00:59:59');
  expect(formatElapsed(3600)).toBe('01:00:00');
  expect(formatElapsed(3661)).toBe('01:01:01');
});

test('clamps negative elapsed time to zero', () => {
  expect(formatElapsed(-5)).toBe('00:00:00');
});
